zipkin-server/src/main/java/zipkin2/server/internal/elasticsearch/ZipkinElasticsearchStorageProperties.java

```diff
@@ -253,6 +262,24 @@ public void setPassword(String password) {
     this.password = emptyToNull(password);
   }
 
+  public String getCredentialsFile() {
+    return credentialsFile;
+  }
+
+  public void setCredentialsFile(final String credentialsFile) {
+    this.credentialsFile = credentialsFile;
+  }
+
+  public Integer getCredentialsRefreshInterval() {
+    return credentialsRefreshInterval;
+  }
+
+  public void setCredentialsRefreshInterval(
+    Integer credentialsRefreshInterval) {
+    // Convert seconds to milliseconds
+    this.credentialsRefreshInterval = credentialsRefreshInterval * 1000;
```

Comment:
this is a surprising hack. please change the place where it is used instead as it doesn't make sense for a property to be manipulated like this. The scheduled expression might accept seconds expression like `s`.

Reply:
I'm sure someone will very soon undo the use of spring scheduling anyway for a normal executor which is simpler and requires no hacks. I don't like adding tech debt especially when converting to the ideal case only takes a few minutes.

Reply:
switch to a ScheduledExecutorService which can be shut down by Spring context.